libs/transactions/src/lib/views/transactions-view/transactions-view.component.ts

```diff
@@ -15,23 +15,35 @@ import { TransactionsHttpService } from '../../services/transactions.http.servic
 export class TransactionsViewComponent {
   public title = this.route.snapshot.data['title'];
   public filter = '';
-
+  public bankAlias$ = this.route.queryParams;
   public transactions$ = combineLatest([
     this.transactionsService.transactions$,
     this.externalCommunicationService?.latestTransaction$ || of(undefined),
+    this.route.queryParams,
   ]).pipe(
-    map(([transactions, latestTransaction]) =>
-      latestTransaction
+    map(([transactions, latestTransaction, routeParameters]) => {
+      const transactionsList = latestTransaction
         ? [latestTransaction, ...(transactions || [])]
-        : transactions
-    )
+        : transactions;
+      if (routeParameters['id']) {
+        return transactionsList?.filter(
+          (item) => item.arrangementId === routeParameters['id']
+        );
+      } else return transactionsList;
+    })
   );
 
   constructor(
     private readonly route: ActivatedRoute,
+    private router: Router,
     private readonly transactionsService: TransactionsHttpService,
     @Optional()
     @Inject(TRANSACTIONS_JOURNEY_COMMUNICATION_SERIVCE)
     private externalCommunicationService: TransactionsCommunicationService
   ) {}
+
+  search(ev: string) {
+    this.filter = ev || '';
+    this.router?.navigate([], { queryParams: { search: this.filter } });
```

Comment:
Don't forget about other current query parameters: `bankAlias` and `id`

Reply:
I added queryParamsHandling: "merge"